issue 15880 added unless method for @types/koa-jwt

winfinit opened the pull request:
added support for unless method

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).


- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.npmjs.com/package/koa-jwt

andy-ms commented:
Please fix the compile error. Try using `type Middleware = Koa.Middleware & { ... }`.
Also, what parameters does the `unless` method take, and what does it return?

winfinit commented:
@andy-ms unless binding is bending pull request in koa-unless repo, when that is going to get merged and published, i can add KoaUnless.Options type. 

what compile error are you getting? it is compiling clean on my side, so i need more details

winfinit commented:
@andy-ms nm, i see issue in travis. thank you.

winfinit commented:
@andy-ms added "unless" options from koa-unless, and thank you for helping on compile error, it is now fixed also.

andy-ms commented on types/koa-jwt/index.d.ts:
```diff
@@ -7,9 +7,11 @@ import Koa = require("koa");
+declare function jwt(options: jwt.Options): jwt.Middleware;
 
 declare namespace jwt {
+    type booleanReturn = (args?: any) => boolean;
```
This doesn't deserve its own alias -- just use `custom?(arg: any): boolean;`
(better, provide a type for `arg`)

winfinit commented:
@andy-ms thank you for reviewing this, added suggested change, and changed arg type to Koa.Context, because based on https://github.com/Foxandxss/koa-unless/blob/master/index.js#L40 that is what is being passed.

andy-ms commented:
Thanks!
